File: fc-chat/demo/part3/ch3/02-6-message-system-add-connection/src/main/java/net/prostars/messagesystem/config/WebSocketHandlerConfig.java
package net.prostars.messagesystem.config;

import net.prostars.messagesystem.auth.WebSocketHttpSessionHandshakeInterceptor;
import net.prostars.messagesystem.handler.WebSocketHandler;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
@SuppressWarnings("unused")
public class WebSocketHandlerConfig implements WebSocketConfigurer {

  private final WebSocketHandler webSocketHandler;
  private final WebSocketHttpSessionHandshakeInterceptor webSocketHttpSessionHandshakeInterceptor;

  public WebSocketHandlerConfig(
      WebSocketHandler webSocketHandler,
      WebSocketHttpSessionHandshakeInterceptor webSocketHttpSessionHandshakeInterceptor) {
    this.webSocketHandler = webSocketHandler;
    this.webSocketHttpSessionHandshakeInterceptor = webSocketHttpSessionHandshakeInterceptor;
  }

  @Override
  public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
    registry
        .addHandler(webSocketHandler, "/ws/v1/message")
        .addInterceptors(webSocketHttpSessionHandshakeInterceptor);
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/config/WebSocketHandlerConfig.kt
package com.yscorp.withpush.messagesystem.config

import com.yscorp.withpush.messagesystem.auth.WebSocketHttpSessionHandshakeInterceptor
import com.yscorp.withpush.messagesystem.handler.WebSocketHandler
import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry

@Configuration
@EnableWebSocket
@Suppress("unused")
class WebSocketHandlerConfig(
    webSocketHandler: WebSocketHandler,
    webSocketHttpSessionHandshakeInterceptor: WebSocketHttpSessionHandshakeInterceptor
) : WebSocketConfigurer {

    private val webSocketHandler: WebSocketHandler = webSocketHandler
    private val webSocketHttpSessionHandshakeInterceptor: WebSocketHttpSessionHandshakeInterceptor =
        webSocketHttpSessionHandshakeInterceptor

    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry
            .addHandler(webSocketHandler, "/ws/v1/message")
            .addInterceptors(webSocketHttpSessionHandshakeInterceptor)
    }

}
